tests/listen_and_simulate.rs

```diff
@@ -21,31 +24,50 @@ fn send_event(event: Event) {
         .expect("Receiving end of EVENT_CHANNEL was closed");
 }
 
-#[test]
-fn test_listen_and_simulate() {
+fn sim_then_listen(events: &mut dyn Iterator<Item = EventType>) -> Result<(), Box<dyn Error>> {
     // spawn new thread because listen blocks
     let _listener = thread::spawn(move || {
         listen(send_event).expect("Could not listen");
     });
+    let tenth_sec = Duration::from_millis(100);
+    thread::sleep(tenth_sec);
```

Comment:
Try waiting for longer ? Maybe the github docker images are slower than this to start the listening ? I have 1 s set on my grab branch to get it working.

Reply:
wow, that seems to have fixed it. Crazy. My rule of thumb is that 100ms is like 10 eternities for a computer. CI must be doing something weird that I'm not thinking of. Maybe Xvfb takes some time to initialize? Or maybe the CI server is virtualized on a host that is very busy. Regardless, I think this is ready to merge.

Reply:
Perfect !